### PR TITLE
Added overload for appSettings configuration to resolve #425

### PR DESCRIPTION
**Pull request by matthewrwilton**
Added a simple overload with the parameters @mivano suggested. I made the index format and template registration optional to remain in the spirit of bare minimum configuration (server address only) required.

If this gets merged in the "Breaking changes for version 2" section in the readme should probably be updated to reflect that basic appSettings configuration is possible.

### CODE REVIEW

File: src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs [hunk 1]
```diff
@@ -56,5 +56,35 @@ public static class LoggerConfigurationElasticsearchExtensions
...
+        /// Overload to allow basic configuration through AppSettings.
+        /// </summary>
+        /// <param name="loggerSinkConfiguration">Options for the sink.</param>
+        /// <param name="uri">A URI for the Elasticsearch instance</param>
+        /// <param name="indexFormat"><see cref="ElasticsearchSinkOptions.IndexFormat"/></param>
+        /// <param name="templateName"><see cref="ElasticsearchSinkOptions.TemplateName"/></param>
+        /// <returns>LoggerConfiguration object</returns>
+        public static LoggerConfiguration Elasticsearch(
+            this LoggerSinkConfiguration loggerSinkConfiguration,
+            string uri,
```
Comment (mivano) [inline src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs]: We can also pass in a Uri directly. This is supported by the AppSettings config (https://github.com/serilog/serilog/blob/dev/src/Serilog/Settings/KeyValuePairs/KeyValuePairSettings.cs#L150).

File: src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs [hunk 2]
```diff
@@ -56,5 +56,35 @@ public static class LoggerConfigurationElasticsearchExtensions
...
+            string indexFormat = null,
+            string templateName = null)
+        {
+            var options = new ElasticsearchSinkOptions(new[] { new Uri(uri) });
```
Comment (mivano) [inline src/Serilog.Sinks.Elasticsearch/LoggerConfigurationElasticSearchExtensions.cs]: Although if we use a string, we might be able to also allow passing in a collection or urls. So do a split on ; or , and pass those in as Uri array.

**Comment by mivano** [comment]
Nice addition. See my inline comments. Not a big deal, but might be worthwhile to consider as changes afterwards are more difficult.

**Comment by matthewrwilton** [comment]
I did wonder about adding support for multiple nodes and since you've brought it up it seems like it is worth adding. I'll add that in and resubmit.

**Comment by matthewrwilton** [comment]
@mivano I added support for configuring multiple nodes using a comma separated list. I also realised that an exception could be thrown if the URI is invalid and I added handling to suppress this (which is how I think Serilog sinks are expected to behave).

**Comment by mivano** [comment]
Thanks. I m just wondering if we should accept an incorrect Uri and silently use the default value when there is an exception. Normally the registration of a sink happens at startup and the other sinks throw an exception (see e.g. https://github.com/serilog/serilog/blob/dev/src/Serilog.FullNetFx/Sinks/RollingFile/RollingFileSink.cs#L64). So a misconfiguration can now lead to a localhost connection. Which might not be there and consequential logging will fail. The selflog is not very discoverable. 

@nblumhardt you have an opinion about this?

Besides this, a nice addition as most people only specify a (collection of) uri(s).

**Comment by nblumhardt** [comment]
I'd favour consistency with the other sinks in this case - keeps things predictable.

**Comment by matthewrwilton** [comment]
No worries. Updated to throw exceptions for invalid URIs.

**Comment by mivano** [comment]
I merged this and a new package will be out soon. Documentation has been updated too. Thanks for the PR!

**Comment by matthewrwilton** [comment]
Happy to help out :smile:

**Pull request merged**